torch_geometric/nn/conv/message_passing.py

```diff
@@ -48,171 +47,131 @@ class MessagePassing(torch.nn.Module):
         node_dim (int, optional): The axis along which to propagate.
             (default: :obj:`0`)
     """
-    __aggr__ = ("add", "mean", "max")
 
-    def __init__(self, aggr="add", flow="source_to_target", node_dim=0):
+    def __init__(self, aggr='add', flow='source_to_target', node_dim=0):
         super(MessagePassing, self).__init__()
 
         self.aggr = aggr
-        assert self.aggr in self.__aggr__
+        assert self.aggr in ['add', 'mean', 'max']
 
         self.flow = flow
-        assert self.flow in ["source_to_target", "target_to_source"]
+        assert self.flow in ['source_to_target', 'target_to_source']
 
         self.node_dim = node_dim
         assert self.node_dim >= 0
 
-        self.__message_signature__ = inspect.signature(self.message)
-        # skip self, out
-        self.__update_signature__ = inspect.signature(self.update)
-        if set(update_special_args) - set(
-            self.__update_signature__.parameters
-        ):
-            raise TypeError(
-                "Incomplete signature of update: {} are "
-                "missing required arguments".format(
-                    (
-                        set(update_special_args)
-                        - set(self.__update_signature__.parameters)
-                    )
-                )
-            )
-        self.__aggregate_signature__ = inspect.signature(self.aggregate)
-        if set(aggregate_special_args) - set(
-            self.__aggregate_signature__.parameters
-        ):
-            raise TypeError(
-                "Incomplete signature of aggregate: {} are "
-                "missing required arguments".format(
-                    set(aggregate_special_args)
-                    - set(self.__aggregate_signature__.parameters)
-                )
-            )
+        self.__msg_sign__ = inspect.signature(self.message).parameters
 
-    def propagate(self, edge_index, size=None, **kwargs):
-        r"""The initial call to start propagating messages.
+        self.__aggr_sign__ = inspect.signature(self.aggregate).parameters
```

Comment:
It is no more signature, but parameters. And why do you pop the first parameter?

Reply:
You are right. I think it is unnecessary to treat the outputs/inputs as special arguments with fixed naming convention and instead inject them as first argument.

Reply:
So what does pop stands for?

Reply:
What do you mean? It just removes the first parameter. This way the output of message and aggregate are not treated as special arguments.

Reply:
Oh, i got you

Reply:
Then it looks fine